### ISSUE TITLE
Cant use registry package in .net standard

### ISSUE THREAD

**Issue by omrycarmel** [dotnet/runtime#51687]
Hi,

I have a .net 4.7.2 framework applications that use and .net framework 4.7.2 libraries.
I'm compiling and running using windows 10 OS.

As a step toward .net core, I converted my libraries target framework to .net standard 2.0

I encountered 2 unexpected behaviors :
1.   if .net framework application has a project reference to the .net standard libraries,
then nugget will copy the .net framework version of the 3rd party libraries that standard library uses,
even though the target framework for my library is .net standard.

This interferes with my  intentions to convert my libraries to .net standard 
because .net core apps wont be able to use this library.

2. In order to work around this issue, I published my .net standard library,
so that nugget will copy the .net standard version of the used packages.
And instead of using project reference, I referenced my .net standard library assembly,
and copied the .net standard 3rd party libraries to the output folder.
But then the Microsoft.Win32.Registry.dll (and maybe other packages as well) throws an exception: "Registry is not supported on this platform."

I'm not sure that its a bug, but according to documentation I expected .net standard 2.0 dlls to work in net472 framework,
And this was not the case for the registry package.

To reproduce:
1. Create a .net standard 2.0 library that use the Microsoft.Win32.Registry package
2. Create a .net framework 4.7.2 console app that use the .net standard library from 1
3. Follow the same steps as above.

**Comment by teo-tsirpanis** [comment]
> if .net framework application has a project reference to the .net standard libraries, then nugget will copy the .net framework version of the 3rd party libraries that standard library uses, even though the target framework for my library is .net standard.

This is not unexpected; .NET Standard is not a framework but a specification without an implementation. When you port your application project to .NET Core I persume it will copy the right libraries.

**Comment by omrycarmel** [comment]
Ok, thanks for the quick answer,

So does that mean that I cant share my .net standard library in a framework application and a core application that are stored in the same folder?

If the answer is yes, then I can just store each application in a different folder with the correct registry library, 
though i thought that it can be more efficient to have only 1 copy of the netstandard2.0  dlls

**Comment by huoyaoyuan** [comment]
The consumed asset is determined by the target framework of entry point, not the transitive library. That's how nuget work.

Imagine this:
```
NetFrameworkExe1
- references NetStandardLibrary1
NetCoreExe1
- references NetStandardLibrary1
NetStandardLibrary1
- references SomePackage

NetFrameworkExe1 will get .net framework version of SomePackage, and NetCoreExe1 will get .net core version of SomePackage.

**Comment by teo-tsirpanis** [comment]
> So does that mean that I cant share my .net standard library in a framework application and a core application that are stored in the same folder?

No, placing them on the same folder seems a bad idea to me. There will be lots of files with the same name, with the most obvious being the `YourApp.exe` assembly for .NET Framework, and the `YourApp.exe` apphost for .NET Core.
